docs/data/data-grid/rows/LazyLoadingGrid.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import {
+  GridFetchRowsParams,
+  DataGridPro,
+  useGridApiRef,
+} from '@mui/x-data-grid-pro';
+import {
+  useDemoData,
+  getRealGridData,
+  getCommodityColumns,
+} from '@mui/x-data-grid-generator';
+
+async function sleep(duration: number) {
+  return new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, duration);
+  });
+}
+
+const loadServerRows = async (newRowLength: number) => {
+  const newData = await getRealGridData(newRowLength, getCommodityColumns());
```

Comment:
Maybe call `getRealGridData` only once, before the grid is loaded, then slice it here to get only the portion needed. This is to avoid replacing already loaded rows with different data.